File: Game/Assets/_Raf/Scripts/WireframeChild.js
private var detail : int = 20;
private var c_render : LineRenderer;
private var dist:float;
private var lightningThickness : float = 1.5;
private var c_wireframeParent:WireframeParent;
public var endPoint : Transform;
public var startPoint:Transform;

private var tempVector:Vector3;
private var player:Transform;
private var c_transform:Transform;


function Start(){
	player=GameObject.FindGameObjectWithTag("Player").transform;
	c_render= GetComponent(LineRenderer);
	c_wireframeParent=transform.parent.gameObject.GetComponent(WireframeParent);
	c_render.material.SetColor("_TintColor", c_wireframeParent.GetColor());
	dist=c_wireframeParent.GetMaxDistance();
	c_render.SetPosition(0,startPoint.position);
	c_render.SetVertexCount(2);
	c_render.SetPosition(1, endPoint.position);
	
	c_render.material.mainTextureScale.x = dist/c_wireframeParent.GetSmoothness();
	c_transform=transform;
}
private var timer:float=0;
function Update () {
	timer+=Time.deltaTime;
	var hit:RaycastHit;
	if(Physics.Raycast(c_transform.position, player.position-c_transform.position, hit,dist)){
		//Debug.DrawRay(transform.position, player.position-transform.position, Color.green);
		if(hit.transform.tag=="Player"){
			if(c_render.enabled!=true) c_render.enabled=true;
			lightningThickness=c_wireframeParent.GetShade();
			c_render.SetWidth(lightningThickness, lightningThickness);
			c_render.material.mainTextureOffset.x = Random.value;
			//positionDistance = dist;
		}else {
			c_render.enabled=false;
			lightningThickness=0;
		}
	}else{
		if(c_render.enabled!=false) c_render.enabled=false;
		lightningThickness=0;
	}
}

File: Game/Assets/_Raf/Scripts/WireFrameChild.cs
using UnityEngine;
using System.Collections;

public class WireFrameChild : MonoBehaviour {
	private float roughness = 0;
	private float thickness = 0;
	private LineRenderer c_lineRenderer;
	private Vector3[] corner;
	public Color color;
	private Transform c_parentTransform;
	private Vector3 tempVector3;
	
	void Start () {
		c_lineRenderer=GetComponent<LineRenderer>();
		c_parentTransform=transform.parent;
		c_lineRenderer.sharedMaterial.SetColor("_TintColor", color);
		c_lineRenderer.SetVertexCount(15);
		corner=new Vector3[15];

		tempVector3.x=-0.5f;
		tempVector3.y=0.5f;
		tempVector3.z=-0.5f;
		corner[0]=tempVector3;
		tempVector3.z=0.4999f;
		corner[1]=tempVector3;
		tempVector3.z=0.5f;
		corner[2]=tempVector3;
		tempVector3.y=-0.4999f;
		corner[3]=tempVector3;
		tempVector3.y=-0.5f;
		corner[4]=tempVector3;
		tempVector3.z=-0.4999f;
		corner[5]=tempVector3;
		tempVector3.z=-0.5f;
		corner[6]=tempVector3;
		tempVector3.y=0.4999f;
		corner[7]=tempVector3;
		tempVector3.y=0.5f;
		corner[8]=tempVector3;
		tempVector3.x=0.4999f;
		corner[9]=tempVector3;
		tempVector3.x=0.5f;
		corner[10]=tempVector3;
		tempVector3.y=-0.4999f;
		corner[11]=tempVector3;
		tempVector3.y=-0.5f;
		corner[12]=tempVector3;
		tempVector3.x=-0.4999f;
		corner[13]=tempVector3;
		tempVector3.x=-0.5f;
		corner[14]=tempVector3;
		for(int i=0 ; i<15 ; i++) c_lineRenderer.SetPosition(i, corner[i]);
	}
	
	void Update () {
		c_lineRenderer.sharedMaterial.mainTextureScale = tempVector3;
		c_lineRenderer.SetWidth(thickness, thickness);
	}
	
	public void SetTextureScale(float textureScale){
		tempVector3.x=textureScale;
	}
	
	public void SetThickness(float thickness){
		this.thickness=thickness;	
	}
}
